gql/server.go

```diff
@@ -58,16 +44,22 @@ func Serve(listenAddr string, dealsDB *db.DealsDB) error {
 	// Allow resolving directly to fields (instead of requiring resolvers to
 	// have a method for every graphql field)
 	opts := []graphql.SchemaOpt{graphql.UseFieldResolvers()}
-	schema, err := graphql.ParseSchema(string(schemaText), newResolver(ctx, dealsDB), opts...)
+	schema, err := graphql.ParseSchema(string(schemaText), newResolver(ctx, s.db, s.provider), opts...)
 	if err != nil {
 		return err
 	}
 
 	// graphQL handler
 	graphQLHandler := graphqlws.NewHandlerFunc(schema, &relay.Handler{Schema: schema})
 
-	fmt.Printf("graphql server listening on %s\n", listenAddr)
-	http.Handle("/graphql", &corsHandler{graphQLHandler})
+	go func() {
+		listenAddr := fmt.Sprintf(":%d", httpPort)
+		fmt.Printf("Graphql server listening on %s\n", listenAddr)
```

Comment:
Nit, but I think we should default to using `log` instead of `fmt.Print` as much as we can.